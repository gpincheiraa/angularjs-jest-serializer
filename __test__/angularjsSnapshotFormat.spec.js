var {Â angularjsSnapshotFormat } = require("../src/angularjsSnapshotFormat");

test("should render a template that doesn't contains AngularJS directives", () => {
    const componentDefinition = {
        template: "<a>Serialize me!</a>",
        $ctrl: {}
    };
    expect(angularjsSnapshotFormat(componentDefinition)).toMatchSnapshot();
});

test("should replace the angular expressions on directives", () => {
    const componentDefinition = {
        template: `
            <div>
                <input ng-model="$ctrl.someValue"> {{ $ctrl.someValue }}
            </div>`,
        $ctrl: {
            someValue : "Hello AngularJS Serializer!"
        }
    };
    expect(angularjsSnapshotFormat(componentDefinition)).toMatchSnapshot();
});

test("should not replace the directives attributes that doesn't have an angular expression", () => {
    const componentDefinition = {
        template: `
            <div ui-view>
                <div class="well well-sm"
                     uib-datepicker
                     ng-model="$ctrl.someValue"
                     datepicker-options="$ctrl.options"></div>
            </div>`,
        $ctrl: {}
    };
    expect(angularjsSnapshotFormat(componentDefinition)).toMatchSnapshot();
});

test(`should overwrite the FormController instance created by angularJS $compile
    function if the component definition set the given variable in the form name attribute`, () => {
        const componentDefinition = {
            template: `
                <form name="$ctrl.formName">
                    <label>Name: </label>
                    <input name="fieldName" ng-model="$ctrl.name" required>
                    <p ng-show="$ctrl.formName.fieldName.$touched">
                        The name field is required.
                    </p>
                </form>
                `,
            $ctrl: {
                formName: {
                    fieldName: {
                        $touched: true
                    }
                }
            }
        };
        //@TODO: In this example the input still remaining the ng-untouched class. This should be changed too.
        expect(angularjsSnapshotFormat(componentDefinition)).toMatchSnapshot();
    });
