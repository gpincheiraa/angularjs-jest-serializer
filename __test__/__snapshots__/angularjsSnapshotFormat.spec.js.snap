// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not replace the directives attributes that doesn't have an angular expression 1`] = `
"<div ui-view=\\"\\" class=\\"ng-scope\\">
                <div class=\\"well well-sm ng-pristine ng-untouched ng-valid ng-empty\\" uib-datepicker=\\"\\" ng-model=\\"[ngExpression]\\" datepicker-options=\\"[ngExpression]\\"></div>
            </div>"
`;

exports[`should overwrite the FormController instance created by angularJS $compile
    function if the component definition set the given variable in the form name attribute 1`] = `
"<form name=\\"$ctrl.formName\\" class=\\"ng-pristine ng-scope ng-invalid ng-invalid-required\\">
                    <label>Name: </label>
                    <input name=\\"fieldName\\" ng-model=\\"[ngExpression]\\" required=\\"\\" class=\\"ng-pristine ng-untouched ng-empty ng-invalid ng-invalid-required\\">
                    <p ng-show=\\"[ngExpression]\\">
                        The name field is required.
                    </p>
                </form>"
`;

exports[`should render a template that doesn't contains AngularJS directives 1`] = `"<a class=\\"ng-scope\\">Serialize me!</a>"`;

exports[`should replace the angular expressions on directives 1`] = `
"<div class=\\"ng-binding ng-scope\\">
                <input ng-model=\\"[ngExpression]\\" class=\\"ng-pristine ng-untouched ng-valid ng-not-empty\\"> Hello AngularJS Serializer!
            </div>"
`;
